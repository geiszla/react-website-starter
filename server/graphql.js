import {
  GraphQLBoolean,
  GraphQLNonNull,
  GraphQLObjectType,
  GraphQLSchema,
  GraphQLString
} from 'graphql';

import bcrypt from 'bcrypt-nodejs';

// Queries
const queryType = new GraphQLObjectType({
  name: 'Query',
  fields: {
    getUsername: {
      type: GraphQLString,
      resolve: ({ session }) => {
        if (session.isLoggedIn) return session.username;
        return null;
      }
    }
  }
});

// Mutations
const mutationType = new GraphQLObjectType({
  name: 'Mutation',
  fields: {
    loginUser: {
      type: new GraphQLNonNull(GraphQLBoolean),
      args: {
        username: { type: new GraphQLNonNull(GraphQLString) },
        password: { type: new GraphQLNonNull(GraphQLString) }
      },
      resolve: ({ session }, { username, password }) => new Promise((resolve) => {
        const correctPassword = '$2a$08$7OBLs4B/PZvSxafCdjIE8.qost5k7QSyS7tUlgPi0ckDpOYYNAhC.'; // testpass

        bcrypt.compare(password, correctPassword, (err, res) => {
          if (err) {
            console.error('Couldn\'t compare password hases.');
            resolve(false);
          }

          if (res === true) {
            session.isLoggedIn = true;
            session.username = username;

            resolve(session.isLoggedIn);
          } else {
            resolve(false);
          }
        });
      })
    },
    logoutUser: {
      type: new GraphQLNonNull(GraphQLBoolean),
      resolve: ({ session }) => {
        session.isLoggedIn = false;
        return session.isLoggedIn;
      }
    }
  }
});

export default new GraphQLSchema({ query: queryType, mutation: mutationType });
