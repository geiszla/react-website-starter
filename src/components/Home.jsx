import React, { Component } from 'react';

import Button from '@material-ui/core/Button';
import PropTypes from 'prop-types';
import Transition from 'react-transition-group/Transition';
import Typography from '@material-ui/core/Typography';
import { observer } from 'mobx-react';
import { withStyles } from '@material-ui/core/styles';

const styles = () => ({
  background: {
    width: '100%',
    height: '100%',

    backgroundImage: 'url("images/in.jpg")',
    backgroundPosition: '35%',
    backgroundRepeat: 'no-repeat',
    backgroundSize: 'cover',
    position: 'fixed',
    filter: 'blur(10px)',

    transition: 'opacity 500ms ease-in-out'
  },
  container: {
    height: '100%',
    transition: 'opacity 500ms ease-in-out'
  },
  content: {
    position: 'relative',
    top: '30%',
    margin: '0 10%',
    textAlign: 'center'
  },
  status: {
    fontSize: '200%',
    color: 'white',
    textAlign: 'center'
  },
  button: {
    marginTop: '50px'
  }
});

const transitionStyles = {
  entering: { opacity: 0 },
  entered: { opacity: 1 }
};

@observer
class Home extends Component {
  render() {
    const {
      classes, fromLogin, username, handleLogout
    } = this.props;

    const initialOpacity = fromLogin ? 0 : 1;
    return (
      <Transition appear={fromLogin} in timeout={0}>
        {state => (
          <div
            className={classes.container}
            style={{ opacity: initialOpacity, ...transitionStyles[state] }}
          >
            <div className={classes.background} style={{ opacity: 1 }} />
            <div className={classes.content}>
              <Typography type="headline" className={classes.status}>
              Hello
                {' '}
                {username}
                {'!'}
              </Typography>
              <Button
                variant="raised"
                className={classes.button}
                onClick={() => handleLogout()}
              >
              Logout
              </Button>
            </div>
          </div>
        )}
      </Transition>
    );
  }
}

Home.propTypes = {
  fromLogin: PropTypes.bool.isRequired,
  username: PropTypes.string.isRequired,
  classes: PropTypes.shape({
    container: PropTypes.string.isRequired,
    background: PropTypes.string.isRequired,
    content: PropTypes.string.isRequired,
    status: PropTypes.string.isRequired,
    button: PropTypes.string.isRequired
  }).isRequired,
  handleLogout: PropTypes.func.isRequired
};

export default withStyles(styles)(Home);
