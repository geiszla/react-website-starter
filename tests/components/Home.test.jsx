import { printSchema } from 'graphql';
import { addMockFunctionsToSchema, makeExecutableSchema } from 'graphql-tools';
import React from 'react';
import TestRenderer from 'react-test-renderer';

import schema from '../../server/graphql';
import Home from '../../src/components/Home.jsx';

const executableSchema = makeExecutableSchema({
  typeDefs: printSchema(schema),
  resolverValidationOptions: { requireResolversForResolveType: false }
});

addMockFunctionsToSchema({ schema: executableSchema });

const handleLogout = jest.fn();
const homeRenderer = TestRenderer.create(
  <Home fromLogin={false} username="username" handleLogout={handleLogout} />
);

describe('Home component', () => {
  it('should match the snapshot when rendered', () => {
    expect(homeRenderer.toJSON()).toMatchSnapshot();

    const fromLoginRenderer = TestRenderer.create(
      <Home fromLogin username="username" handleLogout={handleLogout} />
    );
    expect(fromLoginRenderer.toJSON()).toMatchSnapshot();

    const emptyUsernameRenderer = TestRenderer.create(
      <Home fromLogin username="" handleLogout={handleLogout} />
    );
    expect(emptyUsernameRenderer.toJSON()).toMatchSnapshot();
  });

  it('should call handleLogout when logout button is pressed', () => {
    const logoutButton = homeRenderer.root.find(element => element.type === 'button'
      && element.children
      && element.children.some(child => child.children && child.children[0] === 'Logout'));

    logoutButton.props.onClick();
    expect(handleLogout).toHaveBeenCalled();
  });
});
