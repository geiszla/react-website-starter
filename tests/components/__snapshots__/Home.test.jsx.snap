// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component should match the snapshot when rendered 1`] = `
<div
  className="Home-container-2"
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <div
    className="Home-background-1"
    style={
      Object {
        "opacity": 1,
      }
    }
  />
  <div
    className="Home-content-3"
  >
    <p
      className="MuiTypography-root-6 MuiTypography-body1-15 Home-status-4"
      type="headline"
    >
      Hello
       
      username
      !
    </p>
    <button
      className="MuiButtonBase-root-67 MuiButton-root-41 MuiButton-contained-52 MuiButton-raised-55 Home-button-5"
      disabled={false}
      onBlur={[Function]}
      onClick={[MockFunction]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiButton-label-42"
      >
        Logout
      </span>
      <span
        className="MuiTouchRipple-root-70"
      />
    </button>
  </div>
</div>
`;

exports[`Home component should match the snapshot when rendered 2`] = `
<div
  className="Home-container-2"
  style={
    Object {
      "opacity": 0,
    }
  }
>
  <div
    className="Home-background-1"
    style={
      Object {
        "opacity": 1,
      }
    }
  />
  <div
    className="Home-content-3"
  >
    <p
      className="MuiTypography-root-6 MuiTypography-body1-15 Home-status-4"
      type="headline"
    >
      Hello
       
      username
      !
    </p>
    <button
      className="MuiButtonBase-root-67 MuiButton-root-41 MuiButton-contained-52 MuiButton-raised-55 Home-button-5"
      disabled={false}
      onBlur={[Function]}
      onClick={[MockFunction]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiButton-label-42"
      >
        Logout
      </span>
      <span
        className="MuiTouchRipple-root-70"
      />
    </button>
  </div>
</div>
`;

exports[`Home component should match the snapshot when rendered 3`] = `
<div
  className="Home-container-2"
  style={
    Object {
      "opacity": 0,
    }
  }
>
  <div
    className="Home-background-1"
    style={
      Object {
        "opacity": 1,
      }
    }
  />
  <div
    className="Home-content-3"
  >
    <p
      className="MuiTypography-root-6 MuiTypography-body1-15 Home-status-4"
      type="headline"
    >
      Hello
       
      
      !
    </p>
    <button
      className="MuiButtonBase-root-67 MuiButton-root-41 MuiButton-contained-52 MuiButton-raised-55 Home-button-5"
      disabled={false}
      onBlur={[Function]}
      onClick={[MockFunction]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiButton-label-42"
      >
        Logout
      </span>
      <span
        className="MuiTouchRipple-root-70"
      />
    </button>
  </div>
</div>
`;
